### PR TITLE
Nightingale variation

### PR DESCRIPTION
**Pull request by gustavo-salazar**
Migration to v4: from `protvista-variation` to `@nightingale-elements/nightingale-variation`

The only feature that is not migrated is the use of filters.

### CODE REVIEW

File: packages/nightingale-variation/src/nightingale-variation.ts [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+    nightingale-variation svg {
+      background-color: #fff;
+    }
+
+    nightingale-variation circle {
+      opacity: 0.6;
+    }
+    nightingale-variation circle:hover {
+      opacity: 0.9;
```
Comment (matthiasblum) [inline packages/nightingale-variation/src/nightingale-variation.ts:162]:
```suggestion
      opacity: 1;
```

File: stories/15.Variation/NightingaleVariation.stories.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        sequence=${data.P99999.sequence}
+        highlight-event="onmouseover"
+      ></nightingale-sequence>
+    </div>
+    <nightingale-variation
+      protein-api
+      id="links-2"
+      height="500"
+      length=${data.P99999.sequence.length}
+      highlight="3:20"
```
Comment (matthiasblum) [inline stories/15.Variation/NightingaleVariation.stories.ts:97]: Same comment as with the other story.

**Review by matthiasblum — COMMENTED** [review]
Looks nice, thanks.

I've added a few comments/suggestions in files.

I noted that ticks on the left Y-axis are slightly translated to the left, but maybe a bit too much: the `W` symbol is slightly hidden (I thought it was an italicised _N_ :man_facepalming: ):

image

File: packages/nightingale-variation/src/nightingale-variation.ts [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+  "R",
+  "K",
+  "H",
+  "F",
+  "Y",
+  "W",
+  "P",
+  "d",
+  "*",
+];
```
[the remark this answers is not in the document]
Comment (gustavo-salazar) [reply packages/nightingale-variation/src/nightingale-variation.ts:73]: I have no idea, I took what @xwatkins did for UniProt, as you can see the component in there uses the same order.
BTW, the proteins API does return `*` for `alternativeSequence`. [example](https://www.ebi.ac.uk/proteins/api/variation/P99999) so I imagined it is the same with `d`

File: stories/15.Variation/NightingaleVariation.stories.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+import "../../packages/nightingale-sequence/src/index.ts";
+import "../../packages/nightingale-manager/src/index.ts";
+
+import variationP99999 from "../../packages/nightingale-variation/tests/P99999.variation.json";
+import variationP42336 from "../../packages/nightingale-variation/tests/P42336.variation.json";
+import { ProteinsAPIVariation } from "../../packages/nightingale-variation/src/proteinAPI.js";
+
+const data: Record<string, ProteinsAPIVariation> = {
+  P99999: variationP99999 as unknown as ProteinsAPIVariation,
+  "P99999-2": variationP99999 as unknown as ProteinsAPIVariation,
```
[the remark this answers is not in the document]
Comment (gustavo-salazar) [reply stories/15.Variation/NightingaleVariation.stories.ts:14]: I was testing that reloading the same data doesn't re-renders everything (as when you use it in React). I guess it's a good test for development but not so useful for the showcase app

File: packages/nightingale-variation/src/nightingale-variation.ts [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+  start: number;
+  size?: number;
+  xrefNames: string[];
+  hasPredictions: boolean;
+  tooltipContent?: string;
+  protvistaFeatureId: string;
+  alternativeSequence?: string;
+  internalId?: string;
+  wildType?: string;
+  color?: string;
```
Comment (matthiasblum) [inline packages/nightingale-variation/src/nightingale-variation.ts:35]:
```suggestion
  color?: string;
  consequenceType: string;
```

File: packages/nightingale-variation/src/proteinAPI.ts [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+            ? (variant.genomicLocation || [])[0] 
+            : '',
+        variant: variant.alternativeSequence
+          ? variant.alternativeSequence
+          : AminoAcid.Empty,
+        start: Number(variant.begin),
+        xrefNames: getSourceType(variant.xrefs, variant.sourceType),
+        hasPredictions: variant.predictions && variant.predictions.length > 0,
+        //   tooltipContent: formatTooltip(variant),
+        protvistaFeatureId: String(Math.random()),
```
Comment (matthiasblum) [inline packages/nightingale-variation/src/proteinAPI.ts:231]:
```suggestion
        protvistaFeatureId: String(Math.random()),
        consequenceType: variant.consequenceType,
```

File: packages/nightingale-variation/src/nightingale-variation.ts [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+    this["margin-top"] = 10;
+    this["margin-bottom"] = 10;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    // scale for Amino Acids
+    this.yScale = scalePoint()
+      .domain(aaList)
```
[the remark this answers is not in the document]
Comment (gustavo-salazar) [reply packages/nightingale-variation/src/nightingale-variation.ts:125]: sure I can work on that.

File: packages/nightingale-variation/src/proteinAPI.ts [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+  Empty = "*",
+  F = "F",
+  G = "G",
+  H = "H",
+  I = "I",
+  K = "K",
+  L = "L",
+  M = "M",
+  N = "N",
+  Nl = "NL",
```
[the remark this answers is not in the document]
Comment (gustavo-salazar) [reply packages/nightingale-variation/src/proteinAPI.ts:56]: no idea!
@aurel-l do you know?

File: stories/15.Variation/NightingaleVariation.stories.ts [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+};
+
+export default {
+  title: "Components/Tracks/Variation",
+  argTypes: {
+    protein: {
+      options: ["P99999", "P99999-2", "P42336"],
+      control: { type: "radio" },
+    },
+  },
```
[the remark this answers is not in the document]
Comment (gustavo-salazar) [reply stories/15.Variation/NightingaleVariation.stories.ts:25]: I'm moving this to only be on the one story that needs it.

**Comment by gustavo-salazar** [comment]
> I noted that ticks on the left Y-axis are slightly translated to the left, but maybe a bit too much: the `W` symbol is slightly hidden (I thought it was an italicised _N_ 🤦‍♂️ ):

OK so checking how it works, it uses the margin-left to give enough space for the labels, however if we only change it in the component, then it gets misaligned with other components. so for the story I have adjusted the margin on the other components but then it means that when using it somewhere else, like in the interpro website we are going to update that margin for all the components as well.

File: packages/nightingale-variation/src/nightingale-variation.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (gustavo-salazar) [reply packages/nightingale-variation/src/nightingale-variation.ts:139]: Ok I gave this a try and it was quite simple, so just added a commit that includes a new attribute `row-height`, which has priority over `height` when calculating the space of each row.

**Review by matthiasblum — APPROVED** [review]
Thanks for the changes. Looks great.

**Pull request merged**